### DISCUSSION TITLE
md5 variable injection always invalid

### DISCUSSION THREAD

**Discussion by jeusdi** [Orange-OpenSource/hurl#2581] [Q&A]
I'm trying to make this request using an external variable injection:

```
GET {{get-presigned-url}}
HTTP 200
[Asserts]
md5 == {{md5}}
```
where md5 is:

```
md5="hex,a6474a1b1a0795c4891ba2962543361f;"
```

I've also tried with: `md5 == hex,{{md5}};` assertion and variable `md5=a6474a1b1a0795c4891ba2962543361f`.

Any ideas?

**Comment by jcamiel** [comment]
Hi @jeusdi 

Looking at the [Hurl grammar](https://hurl.dev/docs/grammar.html#oneline-hex), we don't support templatizing hex strings. We can do an issue on it, to support it a future release.

**Reply by jeusdi** [reply]
Which workaround do you suggest me in order to avoid to hardcode md5 checksum?

**Reply by jcamiel** [reply]
The only think I can think of is using a file for the value. For instance, something like this could work (I've not tested it):

```hurl
GET {{get-presigned-url}}
HTTP 200
[Asserts]
md5 == file,{{md5}};
```


where md5 variable points to a local filename whose bytes contents is the expected md5....
